core/store/ChainStore/ChainStore.go

```diff
@@ -0,0 +1,996 @@
+
+	log.Debug(fmt.Sprintf("asset key: %x\n", assetKey))
+
+	// PUT VALUE
+	err := bd.st.Put(assetKey.Bytes(), w.Bytes())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (bd *ChainStore) GetAsset(hash Uint256) (*Asset, error) {
+	log.Debug(fmt.Sprintf("GetAsset Hash: %x\n", hash))
+
+	asset := new(Asset)
+
+	prefix := []byte{byte(ST_Info)}
+	data, err_get := bd.st.Get(append(prefix, hash.ToArray()...))
+
+	log.Debug(fmt.Sprintf("GetAsset Data: %x\n", data))
+	if err_get != nil {
+		//TODO: implement error process
+		return nil, err_get
+	}
+
+	r := bytes.NewReader(data)
+	asset.Deserialize(r)
+
+	return asset, nil
+}
+
+func (bd *ChainStore) GetTransaction(hash Uint256) (*tx.Transaction, error) {
+	log.Debug()
+	log.Debug(fmt.Sprintf("GetTransaction Hash: %x\n", hash))
+
+	t := new(tx.Transaction)
+	err := bd.getTx(t, hash)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return t, nil
+}
+
+func (bd *ChainStore) getTx(tx *tx.Transaction, hash Uint256) error {
+	prefix := []byte{byte(DATA_Transaction)}
+	tHash, err_get := bd.st.Get(append(prefix, hash.ToArray()...))
+	//log.Debug(fmt.Sprintf("getTx Data: %x\n", tHash))
+	if err_get != nil {
+		//TODO: implement error process
+		//log.Warn("Get TX from DB error")
+		return err_get
+	}
+
+	r := bytes.NewReader(tHash)
+
+	// get height
+	_, err := serialization.ReadUint32(r)
+	//height, err := serialization.ReadUint32(r)
+	if err != nil {
+		return err
+	}
+	//log.Debug(fmt.Sprintf("tx height: %d\n", height))
+
+	// Deserialize Transaction
+	err = tx.Deserialize(r)
+
+	return err
+}
+
+func (bd *ChainStore) SaveTransaction(tx *tx.Transaction, height uint32) error {
+
+	//////////////////////////////////////////////////////////////
+	// generate key with DATA_Transaction prefix
+	txhash := bytes.NewBuffer(nil)
+	// add transaction header prefix.
+	txhash.WriteByte(byte(DATA_Transaction))
+	// get transaction hash
+	txHashValue := tx.Hash()
+	txHashValue.Serialize(txhash)
+	log.Debug(fmt.Sprintf("transaction header + hash: %x\n", txhash))
+
+	// generate value
+	w := bytes.NewBuffer(nil)
+	serialization.WriteUint32(w, height)
+	tx.Serialize(w)
+	log.Debug(fmt.Sprintf("transaction tx data: %x\n", w))
+
+	// put value
+	err := bd.st.Put(txhash.Bytes(), w.Bytes())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (bd *ChainStore) GetBlock(hash Uint256) (*Block, error) {
+	var b *Block = new(Block)
+
+	b.Blockdata = new(Blockdata)
+	b.Blockdata.Program = new(program.Program)
+
+	prefix := []byte{byte(DATA_Header)}
+	bHash, err_get := bd.st.Get(append(prefix, hash.ToArray()...))
+	if err_get != nil {
+		//TODO: implement error process
+		return nil, err_get
+	}
+	//log.Debug(fmt.Sprintf("GetBlock Data: %x\n", bHash))
+
+	r := bytes.NewReader(bHash)
+
+	// first 8 bytes is sys_fee
+	_, err := serialization.ReadUint64(r)
+	//sysfee, err := serialization.ReadUint64(r)
+	if err != nil {
+		return nil, err
+	}
+	//log.Debug(fmt.Sprintf("sysfee: %d\n", sysfee))
+
```

Comment:
ditto, now  the code is officially release, all the code be submit should be clean up